### DISCUSSION TITLE
Conversion from safetensor to SavedModel possible?

### DISCUSSION THREAD

**Discussion by C000001** [huggingface/safetensors#241] [Q&A]
Hi, 
I'm looking around to buy a Tensor Processing Unit to speed up a model I'll run in the future. I found one from Google, but they claim that they only support tensorflow lite, which is a format that you can convert to via SavedTensor (and a Keras H5 model, or "concrete functions" whatever they are). 

Is it possible to convert safetensor back to SavedModel? If yes, how?

Sorry, I am not quite there yet inside Machine Learning. I feel like the answer is simple, safetensor has the tensors, I just don't know where to look to actually be sure. 

Relevant links: 
Product I'm looking to get: USB stick with TPU https://coral.ai/products/
Converting models: https://www.tensorflow.org/lite/models/convert/convert_models

**Comment by C000001** [comment]
Okay, so, I just found out that THIS TPU is not really good for anything beyond 1GB big models. It's mostly intended for e.g. face detection or the likes. The ones they use at Google Co-labs are different ones, cloud TPUs. Drats. Back to the drawing board.

**Comment by Narsil** [comment]
Ok, for readers. There is only tensforflow that's able to output SavedModel (afaik).

This library is pure tensors, so what you could do if you had the model, was loading the weights from a given file into an existing tensorflow model and then saving it as a SavedModel.
